bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/TypeTags.java

```diff
@@ -61,7 +61,8 @@ public class TypeTags {
     public static final int ANY_TAG = INVOKABLE_TAG + 1;
     public static final int ENDPOINT_TAG = ANY_TAG + 1;
     public static final int SERVICE_TAG = ENDPOINT_TAG + 1;
-    public static final int ARRAY_TAG = SERVICE_TAG + 1;
+    public static final int CLIENT_TAG = SERVICE_TAG + 1;
```

Comment:
Since this is part of the public API, shouldn't we avoid changes like this? The value of the subsequent tags changes with this, so we should probably add CLIENT_TAG at the end.

Reply:
Do we need CLIENT_TAG anyway? This will also break existing code if someone has checked for the object tag for clients, right?

Reply:
It is cleaner to have CLIENT_TAG, but I'm not sure if we should do it in an update release. cc @warunalakshitha

Reply:
On further thought, we probably should have only OBJECT_TAG, and use flags to differentiate between non-networks objects vs service objects vs client objects. We can't remove `SERVICE_TAG` now either though.

Reply:
We dont hardcode the values right, so changing the value of subsequent tags is ok WDYT?

Reply:
Irrespective of how the value is assigned, what type tags are is an integer value associated with each tag. While it is not the ideal way to do it, a user may have checked for equality by the value of the tag.

For example, to check for an array they could have done
```java
type.getTag() == 32 
```

So this can be a breaking change.

Reply:
Updated with [fb79ee8](https://github.com/ballerina-platform/ballerina-lang/pull/37494/commits/fb79ee83ea126476bbc3ec0fe32f3fe952e6b081)